### ISSUE TITLE
Inmemory Store vs Session Storage

### ISSUE THREAD

**Issue by uurcankaya** [Shopify/shopify_app#191]
Quick question, when shop model generator run here is what we get in shopify_session_repository

```
if Rails.configuration.cache_classes
  ShopifyApp::SessionRepository.storage = SessionStorage
else
  ActionDispatch::Reloader.to_prepare do
    ShopifyApp::SessionRepository.storage = SessionStorage
  end
end
```

I tried on Heroku this way, and database reset itself with dyno change. Is this the setting for Inmemorystore?  Should I change it to?:

```
if Rails.configuration.cache_classes
  ShopifyApp::SessionRepository.storage = Shop
else
  ActionDispatch::Reloader.to_prepare do
    ShopifyApp::SessionRepository.storage = Shop
  end
end
```
 readme file confused me on this

**Comment by christianblais** [comment]
The shop model generator should already have [created a file that sets the `ShopifyApp::SessionRepository.storage` to `Shop`](https://github.com/Shopify/shopify_app/blob/bc83ccb6d556d32dc423d1319b07ada482d4101a/lib/generators/shopify_app/shop_model/shop_model_generator.rb#L18-L20). Did you run `rails generate shopify_app:shop_model`?

**Comment by uurcankaya** [comment]
I did run it and it is what I have at  config/initializers/shopify_session_repository.rb

```
if Rails.configuration.cache_classes
  ShopifyApp::SessionRepository.storage = SessionStorage
else
  ActionDispatch::Reloader.to_prepare do
    ShopifyApp::SessionRepository.storage = SessionStorage
  end
end
```

**Comment by christianblais** [comment]
When everything goes as expected (and it works for me on a brand new install), this is the file you should be getting: https://github.com/Shopify/shopify_app/blob/bc83ccb6d556d32dc423d1319b07ada482d4101a/lib/generators/shopify_app/shop_model/templates/shopify_session_repository.rb

**Comment by christianblais** [comment]
So to answer your question, yes, changing the file to this will fix your dyno problem.

**Comment by uurcankaya** [comment]
I have that other app, which I cannot remember when I generated, it has above setting for  config/initializers/shopify_session_repository.rb and works fine on Heroku with Session Storage. Tested many times, restarted dynos many times, works fine.

Has there been any updates on the gem? Why is other app working fine you think?

I fixed gem to '6.2.0' version on both apps

**Comment by kevinhughes27** [comment]
Yes this was a recent change to try and make it easier to understand. Sounds like you got caught between generator updates sorry about that.
